#!/bin/bash

#This script performs the motif enrichment analysis. (Fig. 3B)

WT_Tag_Directory="XXX" #Specify path to your HOMER Tag Directory of WT H3K27Ac ChIP-seq here.
KO_Tag_Directory="XXX" #Specify path to your HOMER Tag Directory of Crk/Crkl KO H3K27Ac ChIP-seq here.

#Convert superEnhancers.txt into BED format
sed '/^#/d' ${WT_Tag_Directory}/superEnhancers.txt | awk -v 'OFS=\t' '{print $2, $3, $4}' \
> ${WT_Tag_Directory}/superEnhancers.bed

sed '/^#/d' ${KO_Tag_Directory}/superEnhancers.txt | awk -v 'OFS=\t' '{print $2, $3, $4}' \
> ${KO_Tag_Directory}/superEnhancers.bed

#Get intersections between ATAC-seq peaks and SEs
ATAC="XXX" #Specify path to your 'MEF_ATAC_peaks.bed' generated by 'ATAC_mapping.sh' here.
sortBed -i ${ATAC} > sorted_MEF_ATAC_peaks.bed
intersectBed -a ${WT_Tag_Directory}/superEnhancers.bed -b sorted_MEF_ATAC_peaks.bed > ATAC_in_WT_SE.bed
intersectBed -a ${KO_Tag_Directory}/superEnhancers.bed -b sorted_MEF_ATAC_peaks.bed > ATAC_in_KO_SE.bed

#Convert BED into HOMER BED format
bed2pos.pl ATAC_in_WT_SE.bed > ATAC_in_WT_SE.hb
bed2pos.pl ATAC_in_KO_SE.bed > ATAC_in_KO_SE.hb

#Motif enrichment calculation. You can specify the number of threads with "-p" option.
findMotifsGenome.pl ATAC_in_WT_SE.hb mm10 ./Motif_enrichment_results -bg ATAC_in_KO_SE.hb -size given